system/include/webgpu/webgpu.h

```diff
@@ -540,9 +589,17 @@ typedef struct WGPUComputePassDescriptor {
     char const * label;
 } WGPUComputePassDescriptor;
 
-typedef struct WGPUDeviceDescriptor {
+typedef struct WGPUCopyTextureForBrowserOptions {
     WGPUChainedStruct const * nextInChain;
-} WGPUDeviceDescriptor;
+    bool flipY;
+} WGPUCopyTextureForBrowserOptions;
+
+typedef struct WGPUDeviceProperties {
+    bool textureCompressionBC;
+    bool shaderFloat16;
+    bool pipelineStatisticsQuery;
+    bool timestampQuery;
+} WGPUDeviceProperties;
```

Comment:
WGPUCopyTextureForBrowserOptions is a Dawn-specific thing (which, like the testing stuff, should be in a separate header eventually).

WGPUDeviceProperties is an outdated Dawn-specific thing and will be replaced (https://github.com/webgpu-native/webgpu-headers/pull/75). Let's revert this diff (keep the WGPUDeviceDescriptor struct).

Reply:
Done here and in webgpu_cpp, and so I removed all mentions of `CopyTextureForBrowser` (including in `WGPUQueue`)
Also, DeviceDescriptor doesn't exist in webgpu_cpp for some reason

Reply:
ah, that's because it's in webgpu-headers but dawn hasn't implemented it yet (and webgpu_cpp is copied from dawn; there's no upstream version of it yet). We'll get there eventually.